scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -760,4 +728,47 @@ object FormatWriter {
       last
     }
   }
+
+  private val horizontalWhitespaceClass = "[^\\S\\r\\n]"
+
+  private val trailingSpace = Pattern.compile(
+    s"${horizontalWhitespaceClass}+$$",
```

Comment:
Is double `$` matter?

Reply:
because this is an `s"...."` string, simple `$` expects a value to interpolate. i think to get just a `$`, you have to use it twice:

https://docs.scala-lang.org/overviews/core/string-interpolation.html#the-s-string-interpolator